x/nexus/keeper/general_message.go

```diff
@@ -0,0 +1,73 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/axelarnetwork/axelar-core/utils"
+	"github.com/axelarnetwork/axelar-core/utils/key"
+	"github.com/axelarnetwork/axelar-core/x/nexus/exported"
+)
+
+func getGeneralMessageKey(chain exported.ChainName, id string) key.Key {
+	return generalMessagePrefix.
+		Append(key.From(chain)).
+		Append(key.FromStr(id))
+}
+
+// SetNewGeneralMessage sets the given general message
+func (k Keeper) SetNewGeneralMessage(ctx sdk.Context, m exported.GeneralMessage) error {
```

Comment:
```suggestion
func (k Keeper) EnqueueMessage(ctx sdk.Context, m exported.GeneralMessage) error {
```
Or `EnqueueCrossChainMessage`, `General` seems redundant :p

Reply:
Is not a queue though as get by id, `SetMessage` / `SetCrossChainMessage`?

Reply:
Yeah, that sounds good, I was thinking of EnqueueTransfer but the flow isn't quite the same here.